### PR TITLE
test: add unit tests for REE.fetch_production_capacity functionality

### PR DESCRIPTION
**Pull request by PhFariaa**
## Description

This PR adds a new automated test file for the REE parser (electricitymap/contrib/capacity_parsers/REE.py) implementing Modified Condition/Decision Coverage (MC/DC) test cases for the fetch_production_capacity method.

The tests cover all internal logical decisions of the function:

C1 → r.status_code == 200

C2 → item["type"] in MODE_MAPPING

C3 → mode in capacity

Four specific test cases (CT1–CT4) were created to ensure independent evaluation of each condition:

CT1 – Valid request, two items of the same type → values are aggregated (C1=T, C2=T, C3=T)

CT2 – Valid request, first occurrence of a mode → initializes new value (C1=T, C2=T, C3=F)

CT3 – Unmapped type → item ignored (C1=T, C2=F)

CT4 – Non-200 HTTP response → request not processed (C1=F)

These tests ensure full MC/DC coverage of the decision logic inside fetch_production_capacity.

### Preview

New file added:
electricitymap/contrib/capacity_parsers/tests/test_REE.py

### Double check

- [X] I have tested my parser changes locally with `poetry run test_parser "zone_key"`
- [X] I have run `pnpx prettier@2 --write .` and `poetry run format` in the top level directory to format my changes.

### CODE REVIEW

File: electricitymap/contrib/capacity_parsers/tests/test_REE.py [hunk 1]
```diff
@@ -0,0 +1,120 @@
...
+
+
+class FakeSession:
+    def __init__(self, response: FakeResponse):
+        self._response = response
+        self.last_call = None
+
+    def get(self, url, params=None):
+        self.last_call = (url, params)
+        return self._response
```
Comment (VIKTORVAV99) [inline electricitymap/contrib/capacity_parsers/tests/test_REE.py:22]: These should not be faked. There is already mock versions in the request library that should be used for testing.

File: electricitymap/contrib/capacity_parsers/tests/test_REE.py [hunk 2]
```diff
@@ -0,0 +1,120 @@
...
+    # Mapeamento mínimo de tipos → modes
+    monkeypatch.setattr(
+        REE,
+        "MODE_MAPPING",
+        {
+            "solar": "solar",
+            "hydro": "hydro",
+        },
+        raising=False,
+    )
```
Comment (VIKTORVAV99) [inline electricitymap/contrib/capacity_parsers/tests/test_REE.py:41]: Why is this mocked?

Also please use english so other community members and our employees can understand the comments (including me).

**Review by VIKTORVAV99 — CHANGES_REQUESTED** [review]
Thanks for adding the test but please make sure that we don't duplicate mock functionality from the library itself and that all comments are in english.
